cdap-data-fabric/src/main/java/io/cdap/cdap/data2/metadata/lineage/field/FieldLineageTable.java

```diff
@@ -481,4 +484,43 @@ private List<Field<?>> getOperationsKey(long checksum) {
     fields.add(Fields.longField(StoreDefinition.FieldLineageStore.CHECKSUM_FIELD, checksum));
     return fields;
   }
+
+  public List<EndPoint> getEndpoints(String namespaceId, ProgramRunId programRunId) throws IOException {
+    List<Range> multiRanges = new ArrayList<>();
+    multiRanges.add(getNamespaceIncomingRange(namespaceId));
+    multiRanges.add(getNamespaceOutgoingRange(namespaceId));
+    List<EndPoint> result = new ArrayList<>();
+    try (CloseableIterator<StructuredRow> iterator = getEndpointChecksumTable().multiScan(multiRanges,
+                                                                                          Integer.MAX_VALUE)) {
+      while (iterator.hasNext()) {
+        StructuredRow row = iterator.next();
+        ProgramRunId retrievedProgramRunId = GSON.fromJson(row.getString(
+          StoreDefinition.FieldLineageStore.PROGRAM_RUN_FIELD), ProgramRunId.class);
+        if (retrievedProgramRunId.equals(programRunId)) {
```

Comment:
nit: in these cases, it's more readable to check the reverse in order to make it a bit less nested:

```
while (...) {
  if (!...) {
    continue;
  }
  ...
}
```